features/org.wso2.carbon.appmgt.store.feature/src/main/resources/user-portal/controllers/acs.jag

```diff
@@ -53,22 +54,29 @@ var auditLog = require('/modules/auditLog/logger.js');
         sso_sessions = application.get('sso_sessions');
     }
 
-    // Get relay state preserved.
-    var relayState;
-    var relayStates = application.get("relayStates");
-
-    // Send an error if the relay state can't be found.
-    if(!relayStates || !(relayState = relayStates[relayStateId])){
-        log.error('Relay state for ID : ' + relayStateId + " can not be found.");
-        response.sendError(500, 'Cannnot proceed further. Relay state is not present.');
-        return;
+    // Extract tenant domain from request URL.
+    var applicationTenantDomain = util.getTenantDomainFromUrl(request.getRequestURI());
+    if (applicationTenantDomain) {
+        if (log.isDebugEnabled()) {
+            log.debug("User portal ACS accessed from a tenant aware URL for tenant: " + applicationTenantDomain);
+        }
     }
 
-    delete relayStates[relayStateId];
+    // Get relay state preserved.
+    var relayStateURL = session.get(relayStateId);
+    session.remove(relayStateId);
 
-    var redirectURL = String(relayState.requestedURL);
+    var redirectURL = util.getDefaultRedirectUrl(request, applicationTenantDomain);
     if(log.isDebugEnabled()){
-        log.debug("Redirect URL preserved in relay state : " + redirectURL);
+        log.debug("Default redirect URL : " + redirectURL);
+    }
+
+    if (relayStateURL) {
```

Comment:
Can we get default redirect url in the else block of this if condition, instead of always retrieving it?